### ISSUE TITLE
RabbitMQ plugin breaks in swift-only environment.

### ISSUE THREAD

**Issue by cfarquhar** [rcbops/rpc-maas#253]
When running a swift-only control plane, you don't get many (any?) rabbitmq connections, so the plugin fails with this error message:

```
root@infra03-c:/usr/lib/rackspace-monitoring-agent/plugins# /usr/lib/rackspace-monitoring-agent/plugins/run_plugin_in_venv.sh ./rabbitmq_status.py -H 192.168.31.92 -n infra03-c_rabbit_mq_container-c30c2ad3 -U maas_user -p secret
status error get_connection_metrics(session, metrics, options.host, options.port)\n  File "./rabbitmq_status.py", line 115, in _get_connection_metrics\n    max_chans = max(connection['channels'] for connection in response\nValueError: max() arg is an empty sequence\n
Traceback (most recent call last):
  File "./rabbitmq_status.py", line 197, in <module>
    main()
  File "./rabbitmq_status.py", line 180, in main
    _get_connection_metrics(session, metrics, options.host, options.port)
  File "./rabbitmq_status.py", line 115, in _get_connection_metrics
    max_chans = max(connection['channels'] for connection in response
ValueError: max() arg is an empty sequence
```
As a reference point, here are the connections in a more standard (i.e. not swift-only) control plane.  Heat, ironic, neutron, and nova vhosts report connections, while keystone does not.

```
root@infra01-a-rabbit-mq-container-3c9959f0:/# rabbitmqctl list_connections | grep -v 'Listing connections' | awk '{print $1}' | sort | uniq -c
    300 heat
      3 ironic
     95 neutron
    253 nova
```

I verified that the following patch solves the problem, but I wanted feedback from the community that this approach is acceptable before creating a PR.  Can someone confirm they are comfortable with this approach, or suggest a different solution?

```
root@infra02-c:/usr/lib/rackspace-monitoring-agent/plugins# diff -u rabbitmq_status.py{,-patched}
--- rabbitmq_status.py  2017-05-19 07:34:11.931265045 +1000
+++ rabbitmq_status.py-patched  2017-05-19 07:33:59.443264589 +1000
@@ -18,6 +18,7 @@
 import optparse
 import subprocess

+from itertools import chain
 from maas_common import metric
 from maas_common import metric_bool
 from maas_common import print_output
@@ -112,8 +113,10 @@

     response = _get_rabbit_json(session, CONNECTIONS_URL % (host, port))

-    max_chans = max(connection['channels'] for connection in response
-                    if 'channels' in connection)
+# NOTE(cfarquhar): the itertools.chain invocation below is a pre-Python3 hack
+# to default to 0 if we get an empty list of connections back.
+    max_chans = max(chain(connection['channels'] for connection in response
+                    if 'channels' in connection), [0])
     for k in CONNECTIONS_METRICS:
         metrics[k] = {'value': max_chans, 'unit': CONNECTIONS_METRICS[k]}
```

Please note that I will need to get this into the newton and mitaka branches of rpc-openstack.

**Comment by cloudnull** [comment]
Looks good to me, mind submitting a PR w/ the patch so that we can gate the changes, review it, and potentially merge it ?

**Comment by cfarquhar** [comment]
Thanks for the feedback.  PR incoming.

**Comment by cloudnull** [comment]
resolved in #254
